sync/src/relayer/transaction_process.rs

```diff
@@ -77,13 +77,20 @@ impl<'a, CS: ChainStore> TransactionProcess<'a, CS> {
                         .send_message_to(target_peer, fbb.finished_data().to_vec());
                 }
             }
-            Err(PoolError::InvalidTx(TransactionError::Unknown))
-            | Err(PoolError::InvalidTx(TransactionError::Conflict))
-            | Err(PoolError::InvalidTx(TransactionError::Immature))
-            | Err(PoolError::InvalidTx(TransactionError::CellbaseImmaturity)) => {
-                // this error may occured when peer's tip is different with us,
-                // we can't proof peer is bad so just ignore this
-                debug!(target: "relay", "peer {} relay a conflict or unknown input / dep tx: {:?}", self.peer, tx);
+            Err(PoolError::InvalidTx(TransactionError::NullInput))
```

Comment:
This looks ugly. It is better to add a function on `TransactionError` or `PoolError` to tell whether the peer should be banned.